### PR TITLE
Unit tests

### PR DESCRIPTION
**Pull request by miiu96**
## Description of the reasoning behind the pull request (what feature was missing / how the problem was manifesting itself / what was the motive behind the refactoring)
-  Unit tests and fixes

### CODE REVIEW

File: outport/mock/economicsDataMock.go [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+	return core.SafeMul(tx.GetGasPrice(), e.ComputeGasLimit(tx))
+}
+
+// IsInterfaceNil returns true if there is no value under the interface
+func (e *EconomicsHandlerMock) IsInterfaceNil() bool {
+	return e == nil
+}
+
+// ComputeFeeForProcessing will compute the fee using the gas price modifier, the gas to use and the actual gas price
+func (e *EconomicsHandlerMock) ComputeFeeForProcessing(tx coreData.TransactionWithFeeHandler, gasToUse uint64) *big.Int {
```
Comment (sstanculeanu) [inline outport/mock/economicsDataMock.go:90]: no need for this method to be exported

File: outport/mock/economicsDataMock.go [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+	gasPrice := e.GasPriceForProcessing(tx)
+	return core.SafeMul(gasPrice, gasToUse)
+}
+
+// GasPriceForProcessing computes the price for the gas in addition to balance movement and data
+func (e *EconomicsHandlerMock) GasPriceForProcessing(tx coreData.TransactionWithFeeHandler) uint64 {
```
Comment (sstanculeanu) [inline outport/mock/economicsDataMock.go:96]: no need for this one to be exported too

File: outport/mock/economicsDataMock.go [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+		return moveBalanceFee
+	}
+
+	extraFee := e.ComputeFeeForProcessing(tx, difference)
+	moveBalanceFee.Add(moveBalanceFee, extraFee)
+	return moveBalanceFee
+}
+
+// SplitTxGasInCategories returns the gas split per categories
+func (e *EconomicsHandlerMock) SplitTxGasInCategories(tx coreData.TransactionWithFeeHandler) (gasLimitMove, gasLimitProcess uint64) {
```
Comment (sstanculeanu) [inline outport/mock/economicsDataMock.go:113]: same note here, no need for exported

File: outport/mock/alteredAccountsProviderStub.go [hunk 4]
```diff
@@ -0,0 +1,24 @@
+package mock
```
Comment (sstanculeanu) [inline outport/mock/alteredAccountsProviderStub.go:1]: perhaps move this file to testscommon?

File: outport/process/factory/check_test.go [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+	"github.com/ElrondNetwork/elrond-go/outport/process/transactionsfee"
+	"github.com/ElrondNetwork/elrond-go/testscommon"
+	"github.com/ElrondNetwork/elrond-go/testscommon/economicsmocks"
+	"github.com/ElrondNetwork/elrond-go/testscommon/genericMocks"
+	"github.com/ElrondNetwork/elrond-go/testscommon/shardingMocks"
+	"github.com/ElrondNetwork/elrond-go/testscommon/state"
+	"github.com/stretchr/testify/require"
+)
+
+func createArgCreateOutportDataProvider() ArgOutportDataProviderFactory {
```
Comment (sstanculeanu) [inline outport/process/factory/check_test.go:17]:
```suggestion
func createArgOutportDataProviderFactory() ArgOutportDataProviderFactory {
```

File: outport/process/factory/check_test.go [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+		EconomicsData:          &economicsmocks.EconomicsHandlerMock{},
+	}
+}
+
+func TestCheckArgCreateOutportDataProvider(t *testing.T) {
+	t.Parallel()
+
+	arg := createArgCreateOutportDataProvider()
+	arg.AddressConverter = nil
+	require.Equal(t, alteredaccounts.ErrNilPubKeyConverter, checkArgCreateOutportDataProvider(arg))
```
Comment (sstanculeanu) [inline outport/process/factory/check_test.go:38]: perhaps rename the original method to `checkArgOutportDataProviderFactory` as well

**Review by BeniaminDrasovean — DISMISSED** [review]

File: outport/mock/economicsDataMock.go (hunk 1 above)
Comment (miiu96) [reply outport/mock/economicsDataMock.go:90]: done

File: outport/mock/economicsDataMock.go (hunk 2 above)
Comment (miiu96) [reply outport/mock/economicsDataMock.go:96]: done

File: outport/mock/economicsDataMock.go (hunk 3 above)
Comment (miiu96) [reply outport/mock/economicsDataMock.go:113]: done

File: outport/mock/alteredAccountsProviderStub.go (hunk 4 above)
Comment (miiu96) [reply outport/mock/alteredAccountsProviderStub.go:1]: moved

File: outport/process/factory/check_test.go (hunk 5 above)
Comment (miiu96) [reply outport/process/factory/check_test.go:17]: done

File: outport/process/factory/check_test.go (hunk 6 above)
Comment (miiu96) [reply outport/process/factory/check_test.go:38]: done

**Review by sstanculeanu — APPROVED** [review]

**Review by BeniaminDrasovean — APPROVED** [review]

**Pull request merged**